testsuite/tests/uart.rs

```diff
@@ -139,6 +205,10 @@ mod tests {
         }
         let c = nb::block!(rx.read());
         assert!(matches!(c, Err(Error::Overrun)));
+
+        // TODO: This test case takes and splits the serial from state.serial1
+        // but does not put it back. But shouln't it? Right now, it has to be
+        // run after all test cases requiring state.serial1.
```

Comment:
This is a short-coming of the way `split()` is implemented. It consumes `Serial` and generates `Tx` and `Rx`. But there is no way, to re-create `Serial` out of `Tx` and `Rx`. This is needed, so that it can be but back. I have some ideas floating around implementing a `join()` method to circumvent this issue, see: https://github.com/Sh3Rm4n/stm32f3xx-hal/commit/799ee7958ff4fddea3c4889f824c74ed06d75a51 (this does not compile yet)

Until something like this is implement, we have to execute the split test last.

Reply:
Thanks for your explanation!